index.html

```diff
@@ -490,6 +490,170 @@ <h3 id="installation-sec">
+          <a data-lt="presents an install prompt">presenting an automated
+          install prompt</a> until a later time (see
+          <a>BeforeInstallPromptEvent.prompt</a>() method).
+        </p>
+        <p>
+          The <a>BeforeInstallPromptEvent</a> has the following internal slots,
+          which are set when the event is <a data-lt=
+          "construct a BeforeInstallPromptEvent">constructed</a>:
+        </p>
+        <dl>
+          <dt>
+            <dfn>[[\didPrompt]]</dfn>
+          </dt>
+          <dd>
+            A boolean, initially <code>false</code>. Represents if this event
+            was used to <a>present an install prompt</a> to the end-user.
+          </dd>
+          <dt>
+            <dfn>[[\promptOutcome]]</dfn>
+          </dt>
+          <dd>
+            A <a>AppBannerPromptOutcome</a> enum value, initially set to
+            <code>null</code>. Represents the outcome of <a>presenting an
+            install prompt</a>.
+          </dd>
+          <dt>
+            <dfn>[[\userResponsePromise]]</dfn>
+          </dt>
+          <dd>
+            A promise, which resolves with an <a>PromptResponseObject</a>,
+            which represent the outcome of <a>presenting an install prompt</a>.
+          </dd>
+        </dl>
+        <section>
+          <h4>
+            Constructor
+          </h4>
+          <p>
+            To construct an instance of the <a>BeforeInstallPromptEvent</a>
+            interface, run the <dfn data-lt=
+            "construct a BeforeInstallPromptEvent">steps to construct a
+            <code>BeforeInstallPromptEvent</code> event</dfn>:
+          </p>
+          <ol>
+            <li>Let <a>[[\didPrompt]]</a> be <code>false</code>.
+            </li>
+            <li>Let <a>[[\userResponsePromise]]</a> be a newly created promise.
+            </li>
+            <li>Let <a>[[\promptOutcome]]</a> be <code>null</code>.
+            </li>
+            <li>If <code>eventInit.userChoice</code> is valid, then:
+              <ol>
+                <li>Resolve <a>[[\userResponsePromise]]</a> with
+                <code>eventInit.userChoice</code>.
+                </li>
+                <li>Set <a>[[\promptOutcome]]</a> to the value of
+                <code>eventInit.userChoice</code>.
+                </li>
+              </ol>
+            </li>
+          </ol>
+        </section>
+        <section>
+          <h4>
+            <code>prompt()</code> method
+          </h4>
+          <p>
+            The <dfn>prompt</dfn> method, when called, runs the following
+            steps:
+          </p>
+          <ol>
+            <li>Let <var>p</var> be a newly created promise.
+            </li>
+            <li>Run the following steps <a>in parallel</a>:
+              <ol>
+                <li>If <a>[[\promptOutcome]]</a> is not <code>null</code>,
+                resolve <var>p</var> with <a>[[\userResponsePromise]]</a> and
```

Comment:
Ok... still in shock... back to this... 

> Why is p a new promise at all? Why can't we just have prompt() return userResponsePromise?

It avoids exposing the internal promise. Then .prompt() can vend promises that all get resolved when the internal one does.